python/cudf/cudf/core/index.py

```diff
@@ -1411,24 +1466,196 @@ def __cuda_array_interface__(self):
         return self._values.__cuda_array_interface__
 
 
+class BaseNumericIndex(GenericIndex):
+    """Immutable, ordered and sliceable sequence of labels.
+    The basic object storing row labels for all cuDF objects.
+
+    Parameters:
+    -----------
+    data : array-like (1-dimensional)
+    dtype : NumPy dtype,
+            but not used.
+    copy : bool
+        Make a copy of input data.
+    name : object
+        Name to be stored in the index.
+
+    Returns
+    -------
+    Index
+    """
+
+    def __new__(cls, data=None, dtype=None, copy=False, name=None):
+
+        out = Frame().__new__(cls)
```

Comment:
Since `__new__` is a staticmethod, it should be called on the class, not an instance:

```
Frame.__new__(cls)
```

Reply:
Looks like we do this in a few different placed - could we please change all those instances?